### PR TITLE
Added compliant and noncompliant DL case for python_cdk

### PR DESCRIPTION
**Pull request by aastjais**
Add non compliant and compliant samples for python_cdk detectors.

Note: All the test cases have 100% recall and precision.

### CODE REVIEW

File: python_cdk/src/detectors/python-cdk-redshift-backup-enabled/python-cdk-redshift-backup-enabled_non-compliant.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# {fact rule=python-cdk-redshift-backup-enabled@v1.0 defects=1}
+import aws_cdk as cdk
+from aws_cdk.aws_redshift import CfnCluster
+from aws_cdk import  Stack 
+
+class CdkStarterself(cdk.self):
+    def __init__(self, scope: cdk.App, id: str):
+        super(scope, id)
+     
+        # Noncompliant: `automated_snapshot_retention_period` is not set.
```
Comment (bhaaayus) [inline python_cdk/src/detectors/python-cdk-redshift-backup-enabled/python-cdk-redshift-backup-enabled_non-compliant.py:13]: Should be `automated_snapshort_retention_period` is set to 0

File: python_cdk/src/detectors/python_cdk_kms_backing_key_rotation_enabled/python_cdk_kms_backing_key_rotation_enabled_non-compliant.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# SPDX-License-Identifier: MIT-0
+
+# {fact rule=python-cdk-kms-backing-key-rotation-enabled@v1.0 defects=1}
+from aws_cdk.aws_kms import Key, KeySpec 
+import aws_cdk as cdk
+from aws_cdk import Stack
+
+class CdkStarterStack(cdk.Stack):
+        def __init__(self, scope: cdk.App, id: str):
+                super(scope, id)
```
Comment (bhaaayus) [inline python_cdk/src/detectors/python_cdk_kms_backing_key_rotation_enabled/python_cdk_kms_backing_key_rotation_enabled_non-compliant.py:11]: The indentation here is not consistent as compared to the other files. Please fix.

**Review by bhaaayus — CHANGES_REQUESTED** [review]

File: python_cdk/src/detectors/python_cdk_kms_backing_key_rotation_enabled/python_cdk_kms_backing_key_rotation_enabled_non-compliant.py (hunk 2 above)
Comment (bhaaayus) [reply python_cdk/src/detectors/python_cdk_kms_backing_key_rotation_enabled/python_cdk_kms_backing_key_rotation_enabled_non-compliant.py:11]: In general please make sure all files follow the same tab spacing for indents.

File: python_cdk/src/detectors/python-cdk-sqs-missing-encryption/python-cdk-sqs-missing-encryption_compliant.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+# {fact rule=python-cdk-sqs-missing-encryption@v1.0 defects=0}
+import aws_cdk as cdk
+from aws_cdk import aws_sqs as sqs
+
+
+class Stack(cdk.Stack):
+
+    def missing_encryption_compliant(self):
+        # Compliant: encryption present.
```
Comment (bhaaayus) [inline python_cdk/src/detectors/python-cdk-sqs-missing-encryption/python-cdk-sqs-missing-encryption_compliant.py:12]: NIT - Should be `Encryption`

File: python_cdk/src/detectors/python-cdk-sqs-missing-encryption/python-cdk-sqs-missing-encryption_non-compliant.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+# {fact rule=python-cdk-sqs-missing-encryption@v1.0 defects=1}
+import aws_cdk as cdk
+from aws_cdk import aws_sqs as sqs
+
+
+class Stack(cdk.Stack):
+
+    def missing_encryption_noncompliant(self):
+        # Noncompliant: missing encryption.
```
Comment (bhaaayus) [inline python_cdk/src/detectors/python-cdk-sqs-missing-encryption/python-cdk-sqs-missing-encryption_non-compliant.py:12]: NIT - Should be `Missing`

**Review by bhaaayus — CHANGES_REQUESTED** [review]

File: python_cdk/src/detectors/python-cdk-api-logging-disabled/python-cdk-api-logging-disabled_compliant.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+# {fact rule=python-cdk-api-logging-disabled@v1.0 defects=0}
+import aws_cdk as cdk
+from aws_cdk import aws_apigatewayv2
+
+
+class APILoggingDisabled(cdk.Stack):
+
+    def api_logging_disabled_compliant(self):
+        # Compliant: logging present.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-api-logging-disabled/python-cdk-api-logging-disabled_compliant.py:12]: "logging enabled" would be the better comment here.

File: python_cdk/src/detectors/python-cdk-ec-2-security-group-description/python-cdk-ec-2-security-group-description_compliant.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import aws_cdk as cdk
+from aws_cdk import Stack
+from aws_cdk.aws_ec2 import SecurityGroup,Vpc
+
+
+class CdkStarterStack(cdk.Stack):
+    def __init__(self, scope: cdk.App, id: str):
+        super(scope, id)
+
+        # Compliant: SecurityGroup description is set.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-ec-2-security-group-description/python-cdk-ec-2-security-group-description_compliant.py:14]: "Description set for the SecurityGroup"

File: python_cdk/src/detectors/python-cdk-ec-2-security-group-description/python-cdk-ec-2-security-group-description_non-compliant.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+from aws_cdk import Stack
+from aws_cdk.aws_ec2 import SecurityGroup,Vpc
+from aws_cdk import aws_ec2 as ec2
+
+
+class CdkStarterStack(cdk.Stack):
+    def __init__(self, scope: cdk.App, id: str):
+        super(scope, id)
+
+        # Noncompliant: SecurityGroup description contains an empty string.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-ec-2-security-group-description/python-cdk-ec-2-security-group-description_non-compliant.py:15]: "# Noncompliant: The SecurityGroup instantiation includes an empty string as the description."

File: python_cdk/src/detectors/python-cdk-ec2-expose-selective-ports/python-cdk-ec2-expose-selective-ports_compliant.py [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+# {fact rule=python-cdk-ec2-expose-selective-ports@v1.0 defects=0}
+import aws_cdk as cdk
+from aws_cdk.aws_ec2 import CfnSecurityGroupIngress
+
+
+class SelectivePorts(cdk.Stack):
+
+    def exposure_of_sensitive_information_compliant(self):
+        # Compliant: 0.0.0.0/0 range is not used.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-ec2-expose-selective-ports/python-cdk-ec2-expose-selective-ports_compliant.py:12]: "# Compliant: The CfnSecurityGroupIngress instantiation does not use the 0.0.0.0/0 range"

File: python_cdk/src/detectors/python-cdk-ec2-expose-selective-ports/python-cdk-ec2-expose-selective-ports_non-compliant.py [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+# {fact rule=python-cdk-ec2-expose-selective-ports@v1.0 defects=1}
+import aws_cdk as cdk
+from aws_cdk.aws_ec2 import CfnSecurityGroupIngress
+
+
+class SelectivePorts(cdk.Stack):
+
+    def exposure_of_sensitive_information_noncompliant(self):
+        # Noncompliant: 0.0.0.0/0 range is used.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-ec2-expose-selective-ports/python-cdk-ec2-expose-selective-ports_non-compliant.py:12]: "# Noncompliant: The CfnSecurityGroupIngress instantiation uses the 0.0.0.0/0 range"

File: python_cdk/src/detectors/python-cdk-elb-tls-https-listeners-only/python-cdk-elb-tls-https-listeners-only_non-compliant.py [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+# {fact rule=python-cdk-elb-tls-https-listeners-only@v1.0 defects=1}
+import aws_cdk as cdk
+from aws_cdk import Stack, Vpc
+from aws_cdk.aws_elasticloadbalancing import LoadBalancer,LoadBalancingProtocol
+from aws_cdk import aws_elasticloadbalancing as elb
+
+class CdkStarterself(cdk.self):
+  def __init__(self, scope: cdk.App, id: str):
+    super(scope, id)
+    # Noncompliant: `LoadBalancingProtocol` is not set.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-elb-tls-https-listeners-only/python-cdk-elb-tls-https-listeners-only_non-compliant.py:13]: "Noncompliant: The LoadBalancingProtocol is set to tcp for the internal protocol"

File: python_cdk/src/detectors/python-cdk-elb-tls-https-listeners-only/python-cdk-elb-tls-https-listeners-only_compliant.py [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+from aws_cdk import aws_elasticloadbalancing as elb
+
+class CdkStarterself(cdk.self):
+  def __init__(self, scope: cdk.App, id: str):
+    super(scope, id)
+    # Compliant: `LoadBalancingProtocol` is set.
+    LoadBalancer(self, 'rELB1', 
+        vpc= Vpc(self, 'rVPC'),
+    ).add_listener(
+        internal_protocol= LoadBalancingProtocol.ssl,
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-elb-tls-https-listeners-only/python-cdk-elb-tls-https-listeners-only_compliant.py:17]: Compliant: The LoadBalancingProtocol is set to ssl for the both the protocol

File: python_cdk/src/detectors/python-cdk-emr-auth-ec-2-key-pair-or-kerberos/python-cdk-emr-auth-ec-2-key-pair-or-kerberos_non-compliant.py [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+from aws_cdk import Stack
+from aws_cdk.aws_emr import CfnCluster
+from aws_cdk.aws_emr.CfnCluster import InstanceGroupConfigProperty,KerberosAttributesProperty
+from aws_cdk import aws_emr as emr
+	 
+class CdkStarterStack(cdk.Stack):
+    def __init__(self, scope: cdk.App, id: str):
+      super(scope, id)   
+        
+    # Noncompliant: `kerberos_attributes` is not set.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-emr-auth-ec-2-key-pair-or-kerberos/python-cdk-emr-auth-ec-2-key-pair-or-kerberos_non-compliant.py:15]: # Noncompliant: The CfnCluster instantiation does not set the kerberos_attributes property.

File: python_cdk/src/detectors/python-cdk-emr-auth-ec-2-key-pair-or-kerberos/python-cdk-emr-auth-ec-2-key-pair-or-kerberos_compliant.py [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+from aws_cdk import Stack
+from aws_cdk.aws_emr import CfnCluster
+from aws_cdk.aws_emr.CfnCluster import InstanceGroupConfigProperty,KerberosAttributesProperty
+from aws_cdk import aws_emr as emr
+	 
+class CdkStarterStack(cdk.Stack):
+    def __init__(self, scope: cdk.App, id: str):
+      super(scope, id)   
+        
+    # Compliant: `kerberos_attributes` is set.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-emr-auth-ec-2-key-pair-or-kerberos/python-cdk-emr-auth-ec-2-key-pair-or-kerberos_compliant.py:15]: "The CfnCluster instantiation sets the kerberos_attributes property with KerberosAttributesProperty, including kdc_admin_password and realm."

File: python_cdk/src/detectors/python-cdk-emrs-3-access-logging/python-cdk-emrs-3-access-logging_non-compliant.py [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+from aws_cdk.aws_emr import CfnCluster
+from aws_cdk import aws_emr as emr
+
+class CdkStarterself(cdk.self):
+    def __init__(self, scope: cdk.App, id: str):
+      super(scope, id)
+
+      # Noncompliant: `log_uri` is not set.
+      CfnCluster(self, 'rEmrCluster', job_flow_role="jobFlowRole", name ='foo', service_role= 'bar')  
+      CfnCluster(self, 'rEmrCluster', job_flow_role="jobFlowRole", name ='foo',service_role= 'bar')
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-emrs-3-access-logging/python-cdk-emrs-3-access-logging_non-compliant.py:16]: Is this a duplicate code ?

File: python_cdk/src/detectors/python-cdk-emrs-3-access-logging/python-cdk-emrs-3-access-logging_compliant.py [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+# {fact rule=python-cdk-emrs-3-access-logging@v1.0 defects=0}
+import aws_cdk as cdk
+from aws_cdk import Stack
+from aws_cdk.aws_emr import CfnCluster
+from aws_cdk import aws_emr as emr
+
+class CdkStarterself(cdk.self):
+    def __init__(self, scope: cdk.App, id: str):
+      super(scope, id)
+      # Compliant: `log_uri` is set.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-emrs-3-access-logging/python-cdk-emrs-3-access-logging_compliant.py:13]: "# Compliant: The CfnCluster instantiation sets the log_uri property."

File: python_cdk/src/detectors/python-cdk-emrs-3-access-logging/python-cdk-emrs-3-access-logging_non-compliant.py [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+from aws_cdk import Stack
+from aws_cdk.aws_emr import CfnCluster
+from aws_cdk import aws_emr as emr
+
+class CdkStarterself(cdk.self):
+    def __init__(self, scope: cdk.App, id: str):
+      super(scope, id)
+
+      # Noncompliant: `log_uri` is not set.
+      CfnCluster(self, 'rEmrCluster', job_flow_role="jobFlowRole", name ='foo', service_role= 'bar')
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-emrs-3-access-logging/python-cdk-emrs-3-access-logging_non-compliant.py:15]: This looks one duplicate line.

File: python_cdk/src/detectors/python-cdk-emrs-3-access-logging/python-cdk-emrs-3-access-logging_non-compliant.py [hunk 17]
```diff
@@ -0,0 +1,18 @@
...
+import aws_cdk as cdk
+from aws_cdk import Stack
+from aws_cdk.aws_emr import CfnCluster
+from aws_cdk import aws_emr as emr
+
+class CdkStarterself(cdk.self):
+    def __init__(self, scope: cdk.App, id: str):
+      super(scope, id)
+
+      # Noncompliant: `log_uri` is not set.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-emrs-3-access-logging/python-cdk-emrs-3-access-logging_non-compliant.py:14]: "# Noncompliant: The CfnCluster instantiation does not set the log_uri property."

File: python_cdk/src/detectors/python-cdk-neptune-cluster-automatic-minor-version-upgrade/python-cdk-neptune-cluster-automatic-minor-version-upgrade_compliant.py [hunk 18]
```diff
@@ -0,0 +1,17 @@
...
+
+# {fact rule=python-cdk-neptune-cluster-automatic-minor-version-upgrade@v1.0 defects=0}
+import aws_cdk as cdk
+from aws_cdk import Stack
+from aws_cdk.aws_neptune import CfnDBInstance
+
+class CdkStarterStack(cdk.Stack):
+    def __init__(self, scope: cdk.App, id: str):
+      super(scope, id)    
+    # Compliant: `auto_minor_version_upgrade` is set.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-neptune-cluster-automatic-minor-version-upgrade/python-cdk-neptune-cluster-automatic-minor-version-upgrade_compliant.py:12]: "# Compliant: The CfnDBInstance instantiation sets the auto_minor_version_upgrade property to True."

File: python_cdk/src/detectors/python-cdk-neptune-cluster-automatic-minor-version-upgrade/python-cdk-neptune-cluster-automatic-minor-version-upgrade_non-compliant.py [hunk 19]
```diff
@@ -0,0 +1,16 @@
...
+
+# {fact rule=python-cdk-neptune-cluster-automatic-minor-version-upgrade@v1.0 defects=1}
+import aws_cdk as cdk
+from aws_cdk import Stack
+from aws_cdk.aws_neptune import CfnDBInstance
+
+class CdkStarterStack(cdk.Stack):
+    def __init__(self, scope: cdk.App, id: str):
+      super(scope, id)    
+    # Noncompliant: `auto_minor_version_upgrade` is not set.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-neptune-cluster-automatic-minor-version-upgrade/python-cdk-neptune-cluster-automatic-minor-version-upgrade_non-compliant.py:12]: "# Noncompliant: The CfnDBInstance instantiation does not set the `auto_minor_version_upgrade` property."

File: python_cdk/src/detectors/python-cdk-open-search-encrypted-at-rest/python-cdk-open-search-encrypted-at-rest_compliant.py [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+import aws_cdk as cdk
+from aws_cdk import Stack
+from aws_cdk.aws_opensearchservice import EncryptionAtRestOptionsProperty
+from aws_cdk.aws_elasticsearch import CfnDomain as LegacyCfnDomain
+
+class CdkStarterStack(cdk.Stack):
+    def __init__(self, scope: cdk.App, id: str):
+      super(scope, id)
+
+      # Compliant: `encryption_at_rest_options` is set.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-open-search-encrypted-at-rest/python-cdk-open-search-encrypted-at-rest_compliant.py:14]: "# Compliant: The LegacyCfnDomain instantiation sets the encryption_at_rest_options property with EncryptionAtRestOptionsProperty, enabling encryption at rest."

File: python_cdk/src/detectors/python-cdk-rds-non-default-port/python-cdk-rds-non-default-port_compliant.py [hunk 21]
```diff
@@ -0,0 +1,18 @@
...
+# {fact rule=python-cdk-rds-non-default-port@v1.0 defects=0}
+import aws_cdk as cdk
+from aws_cdk.aws_rds import DatabaseCluster, DatabaseClusterEngine
+from aws_cdk import aws_rds as rds
+
+class CdkStarterself(cdk.self):
+  def __init__(self, scope: cdk.App, id: str):
+    super(scope, id)
+    
+    # Compliant: Default port is set.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-rds-non-default-port/python-cdk-rds-non-default-port_compliant.py:13]: "# Compliant: The DatabaseCluster instantiation sets a custom port value of 42, overriding the default port."

File: python_cdk/src/detectors/python-cdk-redshift-backup-enabled/python-cdk-redshift-backup-enabled_non-compliant.py [hunk 22]
```diff
@@ -0,0 +1,24 @@
...
+# {fact rule=python-cdk-redshift-backup-enabled@v1.0 defects=1}
+import aws_cdk as cdk
+from aws_cdk.aws_redshift import CfnCluster
+from aws_cdk import  Stack 
+
+class CdkStarterself(cdk.self):
+    def __init__(self, scope: cdk.App, id: str):
+        super(scope, id)
+     
+        # Noncompliant: `automated_snapshot_retention_period` is set to 0
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-redshift-backup-enabled/python-cdk-redshift-backup-enabled_non-compliant.py:13]: "# Noncompliant: The CfnCluster instantiation sets the automated_snapshot_retention_period property to 0, disabling automated snapshots retention."

File: python_cdk/src/detectors/python-cdk-s3-server-access-logs-disabled/python-cdk-s3-server-access-logs-disabled_compliant.py [hunk 23]
```diff
@@ -0,0 +1,19 @@
...
+
+# {fact rule=python-cdk-s3-server-access-logs-disabled@v1.0 defects=0}
+import aws_cdk as cdk
+from aws_cdk.aws_s3 import Bucket
+
+class CdkStarterself(cdk.self):
+  def __init__(self, scope: cdk.App, id: str):
+    super(scope, id)
+
+    # Compliant: `server_access_logs_bucket` is set.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-s3-server-access-logs-disabled/python-cdk-s3-server-access-logs-disabled_compliant.py:12]: "# Compliant: The Bucket instantiation sets the server_access_logs_bucket property, enabling server access logging to the specified bucket."

File: python_cdk/src/detectors/python-cdk-s3-server-access-logs-disabled/python-cdk-s3-server-access-logs-disabled_non-compliant.py [hunk 24]
```diff
@@ -0,0 +1,15 @@
...
+# {fact rule=python-cdk-s3-server-access-logs-disabled@v1.0 defects=1}
+import aws_cdk as cdk
+from aws_cdk import aws_s3 as s3
+from aws_cdk.aws_s3 import Bucket
+
+class CdkStarterself(cdk.self):
+  def __init__(self, scope: cdk.App, id: str):
+    super(scope, id)
+
+    # Noncompliant: `server_access_logs_bucket` is not set.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-s3-server-access-logs-disabled/python-cdk-s3-server-access-logs-disabled_non-compliant.py:13]: "# Noncompliant: The Bucket instantiation does not set the server_access_logs_bucket property, disabling server access logging."

File: python_cdk/src/detectors/python-cdk-sagemaker-notebook-missing-encryption/python-cdk-sagemaker-notebook-missing-encryption_compliant.py [hunk 25]
```diff
@@ -0,0 +1,21 @@
...
+# {fact rule=python-cdk-sagemaker-notebook-missing-encryption@v1.0 defects=0}
+import aws_cdk as cdk
+from aws_cdk.aws_iam import Role, ServicePrincipal
+from aws_cdk.aws_sagemaker import CfnNotebookInstance
+
+class CdkStarterself(cdk.self):
+  def __init__(self, scope: cdk.App, id: str):
+    super(scope, id)
+
+    # Compliant: `kms_key_id` is set.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-sagemaker-notebook-missing-encryption/python-cdk-sagemaker-notebook-missing-encryption_compliant.py:13]: "# Compliant: The CfnNotebookInstance instantiation sets the kms_key_id property, enabling encryption with the specified KMS key."

File: python_cdk/src/detectors/python-cdk-sagemaker-notebook-missing-encryption/python-cdk-sagemaker-notebook-missing-encryption_non-compliant.py [hunk 26]
```diff
@@ -0,0 +1,19 @@
...
+
+# {fact rule=python-cdk-sagemaker-notebook-missing-encryption@v1.0 defects=1}
+import aws_cdk as cdk
+from aws_cdk.aws_iam import Role, ServicePrincipal
+from aws_cdk.aws_sagemaker import CfnNotebookInstance
+
+class CdkStarterself(cdk.self):
+  def __init__(self, scope: cdk.App, id: str):
+    super(scope, id)
+    # Noncompliant: `kms_key_id` is not set
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-sagemaker-notebook-missing-encryption/python-cdk-sagemaker-notebook-missing-encryption_non-compliant.py:12]: # Noncompliant: The CfnNotebookInstance instantiation does not set the kms_key_id property, potentially leaving data unencrypted.

File: python_cdk/src/detectors/python-cdk-sqs-missing-encryption/python-cdk-sqs-missing-encryption_compliant.py [hunk 27]
```diff
@@ -0,0 +1,15 @@
...
+
+# {fact rule=python-cdk-sqs-missing-encryption@v1.0 defects=0}
+import aws_cdk as cdk
+from aws_cdk import aws_sqs as sqs
+
+
+class Stack(cdk.Stack):
+
+    def missing_encryption_compliant(self):
+        # Compliant: Encryption present.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-sqs-missing-encryption/python-cdk-sqs-missing-encryption_compliant.py:12]: "# Compliant: Enabling encryption using an AWS-managed KMS key."

File: python_cdk/src/detectors/python-cdk-sqs-missing-encryption/python-cdk-sqs-missing-encryption_non-compliant.py [hunk 28]
```diff
@@ -0,0 +1,14 @@
...
+# SPDX-License-Identifier: MIT-0
+
+# {fact rule=python-cdk-sqs-missing-encryption@v1.0 defects=1}
+import aws_cdk as cdk
+from aws_cdk import aws_sqs as sqs
+
+
+class Stack(cdk.Stack):
+
+    def missing_encryption_noncompliant(self):
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python-cdk-sqs-missing-encryption/python-cdk-sqs-missing-encryption_non-compliant.py:11]: "# Noncompliant: The Queue instantiation does not set the encryption property, leaving the queue unencrypted."

File: python_cdk/src/detectors/python_cdk_apigw_execution_logging_enabled/python_cdk_apigw_execution_logging_enabled_non-compliant.py [hunk 29]
```diff
@@ -0,0 +1,15 @@
...
+# {fact rule=python-cdk-apigw-execution-logging-enabled@v1.0 defects=1}
+import aws_cdk as cdk
+from aws_cdk import Stack
+from aws_cdk.aws_apigateway import RestApi
+
+class CdkStarterStack(cdk.Stack):
+    def __init__(self, scope: cdk.App, id: str):
+        super(scope, id)
+
+        # Noncompliant: `logging_level` is not present.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python_cdk_apigw_execution_logging_enabled/python_cdk_apigw_execution_logging_enabled_non-compliant.py:13]: "# Noncompliant: The RestApi instantiation does not set the logging_level property, disabling logging for the API."

File: python_cdk/src/detectors/python_cdk_athena_incomplete_encryption/python_cdk_athena_incomplete_encryption_non-compliant.py [hunk 30]
```diff
@@ -0,0 +1,14 @@
...
+
+# {fact rule=python-cdk-athena-incomplete-encryption@v1.0 defects=1}
+from aws_cdk.aws_athena import CfnWorkGroup
+from aws_cdk import Stack
+import aws_cdk
+
+class AthenaIncompleteEncryption:    
+    def __init__(self):
+
+        # Noncompliant: `work_group_configuration` is not set.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python_cdk_athena_incomplete_encryption/python_cdk_athena_incomplete_encryption_non-compliant.py:12]: # Noncompliant: The CfnWorkGroup instantiation does not set the work_group_configuration property, potentially leaving results unencrypted.

File: python_cdk/src/detectors/python_cdk_kms_backing_key_rotation_enabled/python_cdk_kms_backing_key_rotation_enabled_compliant.py [hunk 31]
```diff
@@ -0,0 +1,18 @@
...
+from aws_cdk.aws_kms import Key, KeySpec 
+import aws_cdk as cdk
+from aws_cdk import Stack
+
+
+class CdkStarterStack(cdk.Stack):
+     def __init__(self, scope: cdk.App, id: str):
+        super(scope, id)        
+        
+        # Compliant: `enable_key_rotation` is set.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python_cdk_kms_backing_key_rotation_enabled/python_cdk_kms_backing_key_rotation_enabled_compliant.py:14]: "# Compliant: The Key instantiation sets the enable_key_rotation property to True, enabling automatic rotation of the KMS key."

File: python_cdk/src/detectors/python_cdk_neptunedb_database_authentication_disabled/python_cdk_neptunedb_database_authentication_disabled_compliant.py [hunk 32]
```diff
@@ -0,0 +1,16 @@
...
+# {fact rule=python-cdk-neptunedb-database-authentication-disabled@v1.0 defects=0}
+import aws_cdk as cdk
+from aws_cdk import Stack
+from aws_cdk.aws_neptune import CfnDBCluster
+
+
+class CdkStarterStack(cdk.Stack):
+    def __init__(self, scope: cdk.App, id: str):
+        super(scope, id)
+        # Compliant: `iam_auth_enabled` is set.
```
Comment (pratik-amazon) [inline python_cdk/src/detectors/python_cdk_neptunedb_database_authentication_disabled/python_cdk_neptunedb_database_authentication_disabled_compliant.py:13]: "# Compliant: The CfnDBCluster instantiation sets the iam_auth_enabled property to True, enabling IAM authentication for the database cluster."

**Review by pratik-amazon — COMMENTED** [review]
Please change the messages as mentioned

**Review by pratik-amazon — APPROVED** [review]

**Review by atonaamz — APPROVED** [review]

**Pull request merged**